api/server.go

```diff
@@ -0,0 +1,59 @@
+// Package server provides a server object that represents the GoCert backend
+package server
+
+import (
+	"crypto/tls"
+	"net/http"
+	"time"
+
+	"github.com/canonical/gocert/internal/certificates"
+)
+
+// loadServerCertificates takes in a certificate and a private key and determines
+// whether to use self signed or given certificates, then returns it in a format
+// expected by the server
+func loadServerCertificates(certificate, key string) (*tls.Certificate, error) {
+	if certificate == "" || key == "" {
+		caCertPEM, caPrivateKeyPEM, err := certificates.GenerateCACertificate()
+		if err != nil {
+			return nil, err
+		}
+		certificate, key, err = certificates.GenerateSelfSignedCertificate(caCertPEM, caPrivateKeyPEM)
```

Comment:
I'm not sure about generating self-signed certificates, I'd rather have the cert/key be mandatory parameters and error out if they are not provided. If this is requested in the future, we can add it then. But let's not add this preemptively.

Reply:
All gone